src/frontend/hooks/useMapStyles.ts

```diff
@@ -0,0 +1,125 @@
+import React from "react";
+import { useFocusEffect } from "@react-navigation/native";
+
+import api from "../api";
+import SettingsContext from "../context/SettingsContext";
+import { MapServerStyleInfo } from "../sharedTypes";
+import { useExperiments } from "./useExperiments";
+import { useMapAvailability, onlineStyleURL } from "./useMapAvailability";
+import { useMapImports } from "./useMapImports";
+
+interface StyleInfoWithImport extends MapServerStyleInfo {
+  isImporting: boolean;
+}
+
+// Randomly generated, but should not change, since this is stored in settings
+// if the user selects one of these "legacy" map styles
+export const DEFAULT_MAP_ID = "487x2pc8ws801avhs5hw58qnxc" as const;
+export const CUSTOM_MAP_ID = "vg4ft8yvzwfedzgz1dz7ntneb8" as const;
+
+type Status = "loading" | "error" | "success";
+type StylesAtLeastOne = [StyleInfoWithImport, ...StyleInfoWithImport[]];
+
+/**
+ * Returns list of available map styles. If the backgroundMaps experiment is not
+ * enabled then this is always a single item - either the default online map or
+ * the custom map is available
+ */
+export function useMapStyles() {
+  const customStyleAvailability = useMapAvailability("custom");
+  const [{ backgroundMaps }] = useExperiments();
+  const activeImports = useMapImports();
+  const [settings, setSettings] = React.useContext(SettingsContext);
+  const [stylesList, setStylesList] = React.useState<StyleInfoWithImport[]>([]);
+  // There is nothing to load if background maps experiment is not enabled
+  const [status, setStatus] = React.useState<Status>(
+    backgroundMaps ? "loading" : "success"
+  );
+
+  useFocusEffect(
+    React.useCallback(() => {
+      if (!backgroundMaps) return;
+      let didCancel = false;
+      api.maps
+        .getStyleList()
+        .then(styles => {
+          if (didCancel) return;
+          const stylesWithImportState = styles.map(s => {
+            // Append importing state so we can filter out importing maps if needed
+            return { ...s, isImporting: !!activeImports[s.id] };
+          });
+          setStylesList(stylesWithImportState);
+          setStatus("success");
+        })
+        .catch(() => didCancel || setStatus("error"));
+
+      return () => {
+        didCancel = true;
+      };
+    }, [activeImports, backgroundMaps])
```

Comment:
i added this deps array in some previous commit because it would otherwise potentially infinitely render. does this look right @gmaclennan ? I think where this falls a little short is when mutative operations from the client occur (e.g. deleting a style), this doesn't really trigger. Could potentially define those actions in this hook and use some dummy state token to manually trigger this, I guess 🤔